Ical.Net/Evaluation/RecurrencePatternEvaluator.cs

```diff
@@ -467,7 +467,8 @@ private List<DateTime> GetYearDayVariants(List<DateTime> dates, RecurrencePatter
                     var date1 = date;
                     yearDayDates.AddRange(pattern.ByYearDay.Select(yearDay => yearDay > 0
                         ? date1.AddDays(-date1.DayOfYear + yearDay)
-                        : date1.AddDays(-date1.DayOfYear + 1).AddYears(1).AddDays(yearDay)));
+                        : date1.AddDays(-date1.DayOfYear + 1).AddYears(1).AddDays(yearDay))
```

Comment:
Maybe this method in general and the changes deserves some comments?

Reply:
Absolutely. Added a comment to explain my changes for now.